### ISSUE TITLE
If I Limited Queue to 1, It has some terrible problem

### ISSUE THREAD

**Issue by wujiangwei** [nervgh/angular-file-upload#640]
------ I limit the queue number to 1
1. When I first selected Image(Called A)
2.Then I selected Image again(Called B,not A)

then I submit img,My server receive Image A,not B

And I think,Can you support auto replace logic or same other amazing methods

Still,I see your doc and demo
If I user clear first(need a delete button),Then no issue
But I think this is not a good method

Waiting ~

**Comment by gabrielliwerant** [comment]
Part of the problem is that "single" upload continually adds files to the queue each time you use it, so if you're attempting to send only one file, you have some complicated logic to figure out which one is the one you need to send. I attempted to work around this locally by clearing the queue after a successful file upload, but this leads to another problem that is evident on the demo page (http://nervgh.github.io/pages/angular-file-upload/examples/simple/):

1. Upload a single file
2. Remove all
3. Upload a single file with the same name

This flow is broken. Nothing is uploaded on step 3 when the file name is the same as the one previously entered. and the "onAfterAddingFile" callback does not fire.

Basically, the flow for using the "single" upload just doesn't seem to work properly. I would expect this feature not to make use of a queue system at all and operate as many current examples of file uploads do on the web when there is only a single file to upload.

**Comment by JakeLunn** [comment]
Even destroying the uploader and recreating it does not allow you to select the same file. You have to reload the page entirely.

**Comment by vaibhav-jain** [comment]
I had the same problem this is what I did. 
Fetched the last uploaded photo as this is what user intended to do.
 `var latestPhoto = uploader.queue[uploader.queue.length - 1];` and then uploaded by  `latestPhoto.upload()`

**Comment by nervgh** [comment]
See [that](https://github.com/nervgh/angular-file-upload/issues/649#issuecomment-230588745).
